controllers/storagecluster/generate.go

```diff
@@ -136,14 +136,32 @@ func generateNameForCephRbdMirror(initData *ocsv1.StorageCluster) string {
 
 // generateCephReplicatedSpec returns the ReplicatedSpec for the cephCluster
 // based on the StorageCluster configuration
-func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string) cephv1.ReplicatedSpec {
+func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string, storageType string) cephv1.ReplicatedSpec {
 	crs := cephv1.ReplicatedSpec{}
 
 	crs.Size = getCephPoolReplicatedSize(initData)
 	crs.ReplicasPerFailureDomain = uint(getReplicasPerFailureDomain(initData))
-	//lint:ignore ST1017 required to compare it directly
-	if "data" == poolType {
-		crs.TargetSizeRatio = .49
+	if poolType == "data" {
+		defaultTargetSizeRatio := 0.49
+		var definedSize float64
+
+		switch storageType {
+		case "block":
```

Comment:
dont we need a case for the nfs and the API changes as well?

Reply:
Acc to the design doc discussion we don't need to expose for nfs block pool or replica-1 block pool.

Reply:
https://docs.google.com/document/d/1BOag-Hm9vpmYnpW76MstJXBdDCyAlYQf0rexpdKZnCA/edit?disco=AAABUdNdABs
But just to confirm again, @travisn do we need to expose a field for the nfs block pool? or may be use the same value in the nfs blokcpool as the default block pool? or the nfs block pool should stay at 0.49?

Reply:
For NFS, are you referring to the `.nfs` CephBlockPool? That is just a metadata pool, so it doesn't need the targetSizeRatio to be set. NFS uses a data pool from CephFS.